Integration/TrustedFormIntegration.php

```diff
@@ -96,82 +99,92 @@ public function doEnhancement(Lead $lead)
+                                $lead->addUpdatedField('trusted_form_expires_at', $data->expires_at);
+                                $persist = true;
+                            }
+
+                            // Set new value for trusted_form_share_url from share_url
+                            if (
+                                !empty($data->share_url)
+                                && $data->share_url !== $lead->getFieldValue('trusted_form_share_url')
+                            ) {
+                                $lead->addUpdatedField('trusted_form_share_url', $data->share_url);
+                                $persist = true;
                             }
-                        }
-                        if (!empty($data->message)) {
-                            $this->logger->info('TrustedForm message with contact '.$lead->getId().' '.$data->message);
-                        }
-                        break 2;
-
-                    case 404:
-                        $message = 'Invalid Certificate: '.$data->message;
-                        break 2;
-
-                    case 401:
-                    case 403:
-                        $message = 'Authentication Failure: '.$data->message;
-                        break 2;
-
-                    case 502:
-                    case 503:
-                        // 100ms delay before retrying.
-                        usleep(100000);
-                        break;
-
-                    default:
-                        $message = "Unrecognized response code: $response->code $data->message";
-                        break 2;
+                            $this->logger->info(
+                                'TrustedForm: Contact '.$identifier.' '.(!$persist ? 'NOT ' : '').'updated. '.(!empty($data->message) ? $data->message : '')
+                            );
+
+                            if (!empty($data->warnings)) {
+                                foreach ($data->warnings as $warning) {
+                                    $this->logger->error('TrustedForm warning with contact '.$identifier.' '.$warning);
+                                }
+                            }
+                            break 2;
+
+                        case 404:
+                            $this->logger->error(
+                                'TrustedForm: Invalid Certificate: '.(!empty($data->message) ? $data->message : '')
+                            );
+                            break 2;
+
+                        case 401:
+                        case 403:
+                            $this->logger->error(
+                                'TrustedForm: Authentication Failure: '.(!empty($data->message) ? $data->message : '')
+                            );
+                            break 2;
+
+                        case 502:
+                        case 503:
+                            $this->logger->error('TrustedForm: Number of requests exceeded.');
```

Comment:
This is not correct. This should log how many attempts...its not exceeded until the loop finishes. This might cause confusion to someone reading the log seeing three of theese in a row...possibly multiple times or in quick succession.

Reply:
If it happens 3 times in a row, then that would mean 3 curl requests in sequence hit 503s. That's significant and I'm ok with that verbose logging, rather than none.

Reply:
I'll add the attempt number in there for clarity.